src/control/lib/spdk/src/nvme_control_common.c

```diff
@@ -68,16 +69,27 @@ register_ns(struct ctrlr_entry *centry, struct spdk_nvme_ns *ns)
 	nentry->ns = ns;
 	nentry->next = centry->nss;
 	centry->nss = nentry;
+
+	ngentry = malloc(sizeof(struct ns_entry));
```

Comment:
Could this be made into a helper function that could then be called twice for the two different lists? Inputs would be the ns, ctrlr, and a double pointer to the ns_entry we're prepending this entry onto.

Reply:
this duplication is going away